client_cli/pytests/src/client_cli/client_cli.py

```diff
@@ -53,6 +59,37 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         """
         self.reset()
 
+    def start_listening_to_events(self, peers_ports):
+        """
+        Initializes listening to events on all peers.
+        """
+        self.transaction_status = {}
+        self.threads = []
+        for port in peers_ports:
+            self.config.update_torii_url(port)
```

Comment:
This part looks like every thread will see only latest call to `update_torii_url`.